### DISCUSSION TITLE
Dynamic OneTimeTask

### DISCUSSION THREAD

**Discussion by valerobles** [kagkarlsson/db-scheduler#520] [Q&A]
I'm using the db-scheduler library (not sb-scheduler for Spring Boot) in my Quarkus application. My goal is to have the service check for any pending or stuck tasks at startup. To achieve this, I create a Scheduler at startup like this:
```
Scheduler.create(dataSource)
            .serializer(KotlinSerializer())
            .enableImmediateExecution()
            .pollingInterval(Duration.ofSeconds(5))
            .commitWhenAutocommitDisabled(true).build()
```

However, I'm encountering an issue where, upon scheduling tasks dynamically at runtime, I receive the following error:

```
Found execution with unknown task-name 'some-task'. Adding it to the list of known unresolved task-names.
Failed to find implementation for task with name 'some-task'. Execution will be excluded from due. The scheduler normally deletes unresolved tasks after 14 days. To handle manually, either delete the execution from the database, or add an implementation for it.
```
From what I understand, this error occurs because the scheduler created at startup is not aware of these dynamically created tasks. My tasks are defined using OneTimeTask, and their implementation varies based on different runtime inputs, making them dynamic.

My Questions:
1. Do I need to create a new Scheduler instance for each dynamically created task?
If so, how should I handle the scenario where I want to check for pending tasks at startup?

2. Is there a way to register these dynamic tasks with the scheduler at runtime so that it can recognize them and not raise the "unknown task-name" error?

3. What is the best practice for managing task implementations that vary based on input, particularly concerning checking for stuck or pending tasks at startup?

Any guidance or solutions for handling dynamic task scheduling and ensuring that all tasks are recognized by the scheduler, especially for recovery of pending tasks, would be greatly appreciated.

**Comment by kagkarlsson** [comment]
1.) 2.) 3.) No, you cannot change implementations (i.e. Tasks) after the scheduler has started. Implementations must be known at startup. If there are different code-paths, then use task-data to control what code-path runs.

**Reply by valerobles** [reply]
Thanks for the response! Is there a way for the scheduler to detect all OneTimeTask at startup similar to how it is done in the `db-scheduler-spring-boot` library?

**Reply by kagkarlsson** [reply]
If you are using another framework there might be a way, otherwise you just have to keep your own list and feed it to the `SchedulerBuilder`
